### DISCUSSION TITLE
DVSA / Imperva - Access Denied - Error 15

### DISCUSSION THREAD

**Discussion by jiwaCh** [ultrafunkamsterdam/undetected-chromedriver#1365] [Q&A]
When accessing driverpracticaltest website was loading fine initially. After a while, it has now lead to Error 15.

My project itself is rather long and hence will be impractical to paste below, but the code below will suffice as a valid example:

user_agent = "Mozilla/5.0 (Windows NT 10.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
driver = webdriver.Chrome(options=options, version_main=113)
driver.execute_cdp_cmd('Network.setUserAgentOverride', {"userAgent":user_agent })
driver.get("https://driverpracticaltest.dvsa.gov.uk/login")
When running the script, I get the error 15

![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/78536809/f08eb536-adfb-4827-a78e-2dd359320850)

But, what's more amazing is, when running the same script on another local machine. No error 15 - can browse normally. I suspect This is temporary and will get banned later after long use.

I have also tested the theory with rotating ip address, with and without user-agent and the same pattern follows.

Will running a script inside a rotating vm (different vm per day) be a solution?

update: I have tired the propose solution by #690 and  #986  and as rightly pointed by #1210 it no longer works

**Comment by luqman-hussain** [comment]
Hi @jiwaCh - had any luck with this or found an alternative approach?

**Reply by joemc200** [reply]
i am receiving the same error message. it works for manual use but not when i try to use selenium:

class WebsiteLogin:
    def __init__(self, dl, logger):
        options = EdgeOptions()
        options.use_chromium = True
        options.add_argument("--disable-blink-features=AutomationControlled")
        options.add_argument("--start-maximized")
        
        # Rotate User Agents
        USER_AGENTS = [
            "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
            "Mozilla/5.0 (Windows NT 10.0; WOW64; rv:45.0) Gecko/20100101 Firefox/45.0",
            "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36",
            # Add more user agents
        ]
        options.add_argument(f"user-agent={random.choice(USER_AGENTS)}")
        
        # Use Proxy
        # options.add_argument("--proxy-server=http://your_proxy:port")
        
        edge_driver_path = r"C:\PATH\msedgedriver.exe"
        self.service = EdgeService(executable_path=edge_driver_path)
        self.driver = webdriver.Edge(service=self.service, options=options)
        self.wait = WebDriverWait(self.driver, 1000)
        self.logger = logger
